### PR TITLE
codec: fix datum convert to float32.

### PR DESCRIPTION
**Pull request by coocood**
The datum was not properly converted to float32 in `convertToFloat` function.

### CODE REVIEW

**Comment by coocood** [comment]
@BusyJay @shenli @zimulala PTAL

File: util/types/datum.go [hunk 1]
```diff
@@ -641,54 +641,54 @@ func (d *Datum) ConvertTo(target *FieldType) (Datum, error) {
...
-		f, err := StrToFloat(d.GetString())
+		f, err = StrToFloat(d.GetString())
 		if err != nil {
 			return ret, errors.Trace(err)
 		}
-		ret.SetFloat64(f)
 	case KindMysqlTime:
-		f, _ := d.GetMysqlTime().ToNumber().Float64()
-		ret.SetFloat64(f)
+		f, _ = d.GetMysqlTime().ToNumber().Float64()
```
Comment (zimulala) [inline util/types/datum.go]: Here, why not consider handling errors ?

Comment (coocood) [inline util/types/datum.go]: The second returned value is not error, it is a bool value exact.

**Comment by shenli** [comment]
LGTM

**Comment by zimulala** [comment]
LGTM

**Pull request merged**